pkg/agents/team6/treaties.go

```diff
@@ -0,0 +1,304 @@
+			// (2) time left on this level = 4, duration = 8 --> shortTermFocus 50%
+			// 		--> The treaty counts as much on this level as for the time after
+			// (3) time left on this level = 1, duration = 100 --> shortTermFocus 1%
+			// 		--> The treaty mostly matters in the long term
+
+			shortTermFocus := float64(estimatedTimeLeft) / float64(t.Duration())
+			benefit = shortTermFocus*shortTermBenefit + (1.0-shortTermFocus)*longTermBenefit
+		}
+
+		// 3. If we benefit from the treaty, accept it
+		return benefit > 0.0
+
+	}
+	// We do not consider other type of treaties than LeaveAmountFood and LeavePercentFood
+	return false
+}
+
+func checkCondition(value int, conditionValue int, conditionOp messages.Op) bool {
+	switch conditionOp {
+	case messages.GT:
+		return value > conditionValue
+	case messages.GE:
+		return value >= conditionValue
+	case messages.EQ:
+		return value == conditionValue
+	case messages.LE:
+		return value <= conditionValue
+	case messages.LT:
+		return value < conditionValue
+	default:
+		return true
+	}
+}
+
+// Returns true if a given treaty applies to an agent by comparing the treaty condition to the agent's state
+func (a *CustomAgent6) conditionApplies(t *messages.Treaty) bool {
+	switch t.Condition() {
+	case messages.HP:
+		return checkCondition(a.HP(), t.ConditionValue(), t.ConditionOp())
+	case messages.Floor:
+		return checkCondition(a.Floor(), t.ConditionValue(), t.ConditionOp())
+	case messages.AvailableFood:
+		return checkCondition(int(a.CurrPlatFood()), t.ConditionValue(), t.ConditionOp())
+	default:
+		return true
+	}
+}
+
+// Returns treaties to propose to other agents, based on the current behaviour and randomness
+func (a *CustomAgent6) constructTreaty() messages.Treaty {
+	// Health levels
+	levels := levelsData{
+		strongLevel:  a.HealthInfo().MaxHP * 3 / 5,
+		healthyLevel: a.HealthInfo().MaxHP * 3 / 10,
+		weakLevel:    a.HealthInfo().WeakLevel,
+		critLevel:    0,
+	}
+	proposedTreaty := messages.NewTreaty(1, a.HealthInfo().MaxHP, 2, 1, 1, 1, int(2*a.reassignPeriodGuess), a.ID())
+	// Altruist and Narcissist do not propose treaties
+	switch a.currBehaviour.string() {
+	case "Collectivist":
+		switch rand.Intn(2) {
+		case 0:
+			// ConditionType, conditionValue, RequestType, requestValue, cop, rop, duration, proposerID
+			proposedTreaty = messages.NewTreaty(1, a.HealthInfo().WeakLevel, 2, 1, 1, 1, int(2*a.reassignPeriodGuess), a.ID())
+		case 1:
+			// Different treaty
+			proposedTreaty = messages.NewTreaty(1, a.HealthInfo().WeakLevel, 2, 1, 1, 1, int(4*a.reassignPeriodGuess), a.ID())
+		}
+
+	case "Selfish":
+		switch rand.Intn(2) {
+		case 0:
+			// ConditionType, conditionValue, RequestType, requestValue, cop, rop, duration, proposerID
+			proposedTreaty = messages.NewTreaty(1, levels.strongLevel, 2, 1, 1, 1, int(2*a.reassignPeriodGuess), a.ID())
+		case 1:
+			// Different treaty
+			proposedTreaty = messages.NewTreaty(1, levels.strongLevel, 2, 1, 1, 1, int(4*a.reassignPeriodGuess), a.ID())
```

Comment:
Up to you here, but I would use the ConditionType and RequestType, not the numbers